File: src/main/java/org/ligi/passandroid/helper/CategoryHelper.java
package org.ligi.passandroid.helper;

import android.graphics.Color;

import org.ligi.passandroid.R;

public class CategoryHelper {

    public final static String[] ALL_CATEGORIES = {"boardingPass", "eventTicket", "coupon", "storeCard", "generic"};

    public final static int getHumanCategoryString(String fromPass) {
        switch (fromPass) {
            case "boardingPass":
                return R.string.boarding_pass;

            case "eventTicket":
                return R.string.category_event;

            case "coupon":
                return R.string.category_coupon;

            case "storeCard":
                return R.string.category_storecard;

            case "generic":
                return R.string.category_generic;

            default:
                return R.string.category_none;
        }

    }

    public final static int getCategoryDefaultBG(String category) {
        switch (category) {
            case "boardingPass":
                return 0xFF3d73e9;

            case "eventTicket":
                return 0xFF9f3dd0;

            case "coupon":
                return 0xFF9ccb05;

            case "storeCard":
                return 0xFFf29b21;

            case "generic":
                return 0xFFea3c48;

            default:
                return Color.WHITE;
        }

    }

    public final static int getCategoryDefaultFG(String category) {
        switch (category) {
            case "boardingPass":
                return Color.BLACK;

            case "eventTicket":
                return Color.BLACK;

            case "coupon":
                return Color.BLACK;

            case "storeCard":
                return Color.BLACK;

            case "generic":
                return Color.BLACK;

            default:
                return Color.BLACK;
        }

    }


    public final static int getCategoryTopImageRes(String fromPass) {
        switch (fromPass) {
            case "boardingPass":
                return R.drawable.cat_bp;

            case "eventTicket":
                return R.drawable.cat_et;

            case "coupon":
                return R.drawable.cat_cp;

            case "storeCard":
                return R.drawable.cat_sc;

            case "generic":
            default:
                return R.drawable.cat_none;

        }
    }


    public final static String getCategoryShortStr(String fromPass) {
        switch (fromPass) {
            case "boardingPass":
                return "BP";

            case "eventTicket":
                return "T";

            case "coupon":
                return "%";

            case "storeCard":
                return "SC";

            case "generic":
            default:
                return "*";

        }
    }


}

File: android/src/main/java/org/ligi/passandroid/functions/CategoryHelper.kt
package org.ligi.passandroid.functions

import android.graphics.Color
import android.support.annotation.ColorInt
import android.support.annotation.DrawableRes
import android.support.annotation.StringRes
import org.ligi.passandroid.R
import org.ligi.passandroid.model.pass.PassType

@StringRes
fun getHumanCategoryString(fromPass: PassType) = when (fromPass) {
    PassType.BOARDING -> R.string.boarding_pass
    PassType.EVENT -> R.string.category_event
    PassType.COUPON -> R.string.category_coupon
    PassType.LOYALTY -> R.string.category_storecard
    PassType.GENERIC -> R.string.category_generic
    PassType.VOUCHER -> R.string.categories_voucher

    else -> R.string.category_none
}


@ColorInt
fun getCategoryDefaultBG(category: PassType) = when (category) {
    PassType.BOARDING -> 0xFF3d73e9
    PassType.EVENT -> 0xFF9f3dd0
    PassType.COUPON -> 0xFF9ccb05
    PassType.LOYALTY -> 0xFFf29b21
    PassType.VOUCHER -> 0xFF2A2727
    PassType.GENERIC -> 0xFFea3c48

    else -> Color.WHITE.toLong()
}.toInt()


@DrawableRes
fun getCategoryTopImageRes(type: PassType) = when (type) {
    PassType.BOARDING -> R.drawable.cat_bp
    PassType.EVENT -> R.drawable.cat_et
    PassType.COUPON -> R.drawable.cat_cp
    PassType.LOYALTY -> R.drawable.cat_sc
    PassType.VOUCHER -> R.drawable.cat_ps
    PassType.GENERIC -> R.drawable.cat_none

    else -> R.drawable.cat_none
}
